packages/fether-react/src/utils/withAccount.js

```diff
@@ -0,0 +1,22 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+
+const AccountAddressFromRouter = withRouter(props =>
+  props.children(props.match.params.accountAddress)
+);
+
+// We don't want to pass the router props to the returned component
+export default Component => initialProps => {
+  return (
+    <AccountAddressFromRouter>
+      {accountAddress => (
+        <Component accountAddress={accountAddress} {...initialProps} />
+      )}
+    </AccountAddressFromRouter>
+  );
```

Comment:
Alternate version:

```javascript
branch(
  ({ match, history }) => !!(match && history), // test if withRouter props are already there
  withProps(...),
  compose(withRouter, mapProps(...))
)
```

Though I'm fine with your version. I think I never really liked the render props syntax in React (https://reactjs.org/docs/render-props.html#using-props-other-than-render), but that's just a matter of taste 😃